internal/controllers/topology/cluster/desired_state.go

```diff
@@ -236,11 +235,16 @@ func (r *Reconciler) computeControlPlane(ctx context.Context, s *scope.Scope, in
 
 	controlPlaneAnnotations := util.MergeMap(topologyMetadata.Annotations, clusterClassMetadata.Annotations)
 
+	nameTemplate := "{{ .cluster.name }}-{{ .random }}"
+	if s.Blueprint.ClusterClass.Spec.ControlPlane.NamingStrategy != nil && s.Blueprint.ClusterClass.Spec.ControlPlane.NamingStrategy.Template != nil {
+		nameTemplate = *s.Blueprint.ClusterClass.Spec.ControlPlane.NamingStrategy.Template
+	}
+
 	controlPlane, err := templateToObject(templateToInput{
 		template:              template,
 		templateClonedFromRef: templateClonedFromRef,
 		cluster:               cluster,
-		namePrefix:            fmt.Sprintf("%s-", cluster.Name),
+		nameGenerator:         names.ControlPlaneNameGenerator(nameTemplate, s.Current.Cluster.Name),
```

Comment:
```suggestion
		nameGenerator:         names.ControlPlaneNameGenerator(nameTemplate, cluster.Name),
```
nit: I guess doesn't really matter but as we build that "shortcut" above